components/quiz.tsx

```diff
@@ -0,0 +1,194 @@
+"use client";
+import React, { useState, useEffect } from 'react';
+import { v4 as uuidv4 } from 'uuid';
+import { saveQuizResponse, getQuizResponse, resetQuizResponse } from '@/utils/indexedDB';
+import Image from 'next/image';
+import { cn } from '@/utils/cn';
+import { buttonVariants } from '@/components/ui/button'
+
+interface QuizProps {
+  quizId: string;
+  question: string;
+  options: string[];
+  correctAnswer: string;
+  hint: string;
+  explanation: string;
+}
```

Comment:
Let's support multi-question quizes:

Quiz:
- quizId
- array of questions

Question:
- questionId
- options
- correctAnswer
- hint
- explanation